community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/log/PhysicalLogFileInformation.java

```diff
@@ -30,17 +31,17 @@
 
     private final PhysicalLogFiles logFiles;
     private final LogHeaderCache logHeaderCache;
-    private final TransactionIdStore transactionIdStore;
+    private final Supplier<Long> lastCommittedId;
     private final LogVersionToTimestamp logVersionToTimestamp;
 
     public PhysicalLogFileInformation( PhysicalLogFiles logFiles,
                                        LogHeaderCache logHeaderCache,
-                                       TransactionIdStore transactionIdStore,
+                                       Supplier<Long> lastCommittedId,
```

Comment:
`LastEntryInLog` makes no difference to constructor callers because they will just pass in a lambda or a method reference that takes no arguments and return long.
So `PhysicalLogFileInformation`s with completely different `LastEntryInLog`s can be used interchangeably which most likely is not smth desirable.